packages/transformer/src/IModelCloneContext.ts

```diff
@@ -6,30 +6,64 @@
+  PrimitiveTypeCode, PropertyMetaData, RelatedElement, RelatedElementProps,
 } from "@itwin/core-common";
 import {
-  ElementAspect, EntityReferences, IModelElementCloneContext, SQLiteDb,
+  Element, ElementAspect, EntityReferences, IModelElementCloneContext, IModelJsNative, SQLiteDb,
 } from "@itwin/core-backend";
 import { ECReferenceTypesCache } from "./ECReferenceTypesCache";
 import { EntityUnifier } from "./EntityUnifier";
+import { TransformerLoggerCategory } from "./TransformerLoggerCategory";
+
+const loggerCategory: string = TransformerLoggerCategory.IModelCloneContext;
 
 /** The context for transforming a *source* Element to a *target* Element and remapping internal identifiers to the target iModel.
  * @beta
  */
 export class IModelCloneContext extends IModelElementCloneContext {
 
   private _refTypesCache = new ECReferenceTypesCache();
+  private _aspectRemapTable = new Map<Id64String, Id64String>();
 
   /** perform necessary initialization to use a clone context, namely caching the reference types in the source's schemas */
   public override async initialize() {
     await this._refTypesCache.initAllSchemasInIModel(this.sourceDb);
   }
 
-  private _aspectRemapTable = new Map<Id64String, Id64String>();
+  /** Clone the specified source Element into ElementProps for the target iModel.
+   * @internal
+   */
+  public override cloneElement(sourceElement: Element, cloneOptions?: IModelJsNative.CloneElementOptions): ElementProps {
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    const targetElementProps: ElementProps = this["_nativeContext"].cloneElement(sourceElement.id, cloneOptions);
+    // Ensure that all NavigationProperties in targetElementProps have a defined value so "clearing" changes will be part of the JSON used for update
+    sourceElement.forEachProperty((propertyName: string, meta: PropertyMetaData) => {
+      if ((meta.isNavigation) && (undefined === (sourceElement as any)[propertyName])) {
+        (targetElementProps as any)[propertyName] = RelatedElement.none;
+      }
+    }, false); // exclude custom because C++ has already handled them
+    if (this.isBetweenIModels) {
+      // The native C++ cloneElement strips off federationGuid, want to put it back if transformation is between iModels
+      targetElementProps.federationGuid = sourceElement.federationGuid;
+      const targetElementCodeScopeType = this.targetDb.codeSpecs.getById(targetElementProps.code.spec).scopeType;
+      if (CodeScopeSpec.Type.Repository === targetElementCodeScopeType && targetElementProps.code.scope !== IModel.rootSubjectId) {
+        Logger.logWarning(loggerCategory, `Incorrect CodeScope '${targetElementCodeScopeType}' is set for target element ${targetElementProps.id}`);
+      }
+    }
+    // unlike other references, code cannot be null. If it is null, use an empty code instead
+    if (targetElementProps.code.scope === Id64.invalid || targetElementProps.code.spec === Id64.invalid) {
```

Comment:
note that this is an override that mostly copies the existing stuff in `IModelElementCloneContext` in core. I told @KyryloVolotovskyi it's OK to override here because I am reworking usage of that class extensively right now.

The reason that exists in `IModelElementCloneContext` and is copied here is because it's possible for the transformer to try transforming an element before it has seen its scoping element (e.g. in RelatedElement scope kind), in which case it will insert the element anyway, with an empty code, and queue an UPDATE on this element for later, after it has transformed the scoping element into the target, with the correct code

Reply:
ok, a better explanation in the comment will help (that the code will be preserved later).

Reply:
@KyryloVolotovskyi please do add a comment:
```
// unlike other references, code cannot be null. If it is null, use an empty code instead
// this will be updated back later as the transformer resolves references

Reply:
@MichaelBelousov just added